### PR TITLE
Allow cancellation of work to cause ProcessFiles to return.

### PR DESCRIPTION
**Pull request by kjsanger**
Do not close the ClientPool, so allow running work to complete.

### CODE REVIEW

File: valet/pathproc.go [hunk 1]
```diff
@@ -70,7 +70,14 @@ func ProcessFiles(cancelCtx context.Context, params ProcessParams) {
 		os.Exit(1)
 	}
```
Comment (sb10) [inline valet/pathproc.go]: I haven't gone through and really understood the flow of things, but would DoProcessFiles() not benefit from also responding to the cancelled context? With the PR as it is, do you see the final summary messages from DoProcessFiles() on ctrl-c?

Comment (kjsanger) [reply valet/pathproc.go]: You're right. I should have removed that `os.Exit()` in the goroutine and moved the remaining `os.Exit()` until after the remaining work has come thru the channel.

File: valet/pathproc.go [hunk 2]
```diff
@@ -42,35 +43,41 @@ type ProcessParams struct {
...
+		case <-cancelCtx.Done():
+			log.Info().Msg("processing cancelled")
+			break done
+		}
 	}
 
-	<-done
+	merr := <-merrs
+
+	return utilities.CombineErrors(err, merr)
```
Comment (sb10) [inline valet/pathproc.go]: My understanding is when the context is cancelled, we do not wait for DoProcessFiles to complete, which means err will always be nil. DoProcessFiles needs to take and do something with the context as well, doesn't it?

File: valet/pathproc.go [hunk 3]
```diff
@@ -42,35 +43,41 @@ type ProcessParams struct {
...
+			log.Info().Msg("processing done")
+			break done
+		case <-cancelCtx.Done():
+			log.Info().Msg("processing cancelled")
+			break done
+		}
 	}
 
-	<-done
+	merr := <-merrs
```
Comment (sb10) [inline valet/pathproc.go]: Can there be more than 1 err in merrs? Loop to get them all?

File: valet/pathproc.go [hunk 4]
```diff
@@ -42,35 +43,41 @@ type ProcessParams struct {
 	MaxProc       int           // The maximum number of threads to run
 }
 
-func ProcessFiles(cancelCtx context.Context, params ProcessParams) {
+func ProcessFiles(cancelCtx context.Context, params ProcessParams) error {
 
 	wpaths, werrs := WatchFiles(cancelCtx, params.Root, params.MatchFunc,
```
Comment (sb10) [inline valet/pathproc.go]: WatchFiles and FindFilesInterval use cancelCtx to stop looping, but don't close their output channels. Doesn't that mean that DoProcessFiles just keeps waiting on cancellation?

Comment (sb10) [reply valet/pathproc.go]: Actually, they do close

File: valet/pathproc.go (hunk 2 above)
Comment (kjsanger) [reply valet/pathproc.go]: Changed to always wait for completion on cancel.

**Pull request merged**